### ISSUE TITLE
Can't initialize keras.Model based network with tf.train.init_from_chekpoint

### ISSUE THREAD

**Issue by jrabary** [tensorflow/tensorflow#22298]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**:
- **TensorFlow version (use command below)**: git master
- **Python version**: 3.6
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"

### Describe the problem

We are trying use `tf.keras.Model` as a base class for our neural network. So far, with `slim` based network we have used `tf.train.init_from_checkpoint` to warm start the training.
But, it seems that the current version of `tf.train.init_from_checkpoint` doesn't play well with `keras.Model`. Probably because under the hood, `tf.train.init_from_checkpoint` calls `variable_scope._get_defautl_variable_store()` which doesn't know variables created with keras objects.

Below a snippet of code to illustrate this problem

### Source code / logs

```
import tensorflow as tf

from tensorflow.python.ops import variable_scope as vs

net = tf.keras.Sequential([
    tf.keras.layers.Dense(10, name='fc1')
])

x = tf.random_uniform([10, 3])
y = net(x)

# all variable are created 
vars = tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES)
for v in vars:
    print(v.name)

# but variable scope doesn't see them
# then no variable to initialize for tf.train.init_from_checkpoint
var_store = vs._get_default_variable_store()._vars
print(var_store)
```

**Comment by harshini-gadige** [comment]
@jrabary Hi, are you getting any error or does the system not respond without any error? Could you post the error message if you get it.

**Comment by jrabary** [comment]
@harshini-gadige Hi, no there is no error. But the variable are not initialized. I know this because I set the logging to DEBUG in order to see which variables are initialized in the logs.
The good news is that I can initialize my network tf.estimator.WarmStartSettings.

**Comment by harshini-gadige** [comment]
@fchollet  Any suggestions or explanation on why the network cannot be initialized with tf.train.init_from_checkpoint

**Comment by facaiy** [comment]
@jrabary  How about use the native methods of `tf.keras.Model`, say `load`, `load_weights` or `set_weights`?

**Comment by jrabary** [comment]
@facaiy Need to figure out how to use `load_weights` since we are using tf.Estimator to optimize the model. This is because we also have `old` models coded with slim so we just don't want to change everything.

**Comment by facaiy** [comment]
I'm afraid that the graph built by tf.Estimator is quite different from the one built by keras, so you have to resolve those details by yourself.  Even if tf.train.init_from_checkpoint could support tf.keras, it still won't work as you expected.

If you really need it, I think you can load all variables with ` tf.train.load_variable`, and then assign them one by one with `set_weigths` method of your keras model. By the way, you can get all variable by `get_weights` method.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by DELTA37** [comment]
It is a big problem
MNIST tutorial with tf.Estimator says us to use keras Model for architecture creation task
But if we wanna restore and fine tune tf Estimator, the best way to do it is tf.train.init_from_checkpoint
And it is troubling situation, when you train model and waste time to restore

**Comment by DELTA37** [comment]
I always check that error is within tensorflow
tf.python.ops.variable_scope has a function `variable()`
Which has a strange initializer after creation

**Comment by DELTA37** [comment]
https://github.com/tensorflow/tensorflow/pull/23110
Fixed this problem as commit
It requires testing all framework with this updater
My local tests has been completed

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by sushreebarsa** [comment]
@jrabary Is this still an issue?
I tried to replicate the issue and faced AttributeError. TF version 1.x is no longer actively supported so we recommend you to kindly  upgrade  to latest TF versions,please refer to this [link](https://www.tensorflow.org/guide/migrate).
Thanks!
